Scripts/Flag_placer.py

```diff
@@ -0,0 +1,352 @@
+
+    :return flags: a 2d list containing all the flag information.
+    :return isbuildingflags: a list indicating which flags are currently being built.
+    :return all_flags: A 2d list containing the coordinates of the flags and additional information.
+    """
+    insert_size = abs(read.isize)
+
+    if insert_size > args.high_insert_size:
+        flags, isbuildingflags = generate_flag(read, flags, isbuildingflags, 1)
+        flags[1][3]['lengths'].append(insert_size)
+
+    elif not insert_size > args.high_insert_size and isbuildingflags[1] and start > flags[1][2]:
+        percentage = round(flags[1][3]['count'] / flags[1][3]['total'], 2)
+        if flags[1][3]['count'] > args.threshold and percentage > args.minpercentage:
+            all_flags.append(flags[1])
+        flags[1] = [chromosome, None, None, {'type': 'high_insert_size', 'count': 0, 'total': 0, 'lengths': []}]
+        isbuildingflags[1] = False
+
+    return flags, isbuildingflags, all_flags
+
+
+def flag_unmapped_mate(read, flags, isbuildingflags, all_flags, chromosome, start):
+    """ The flag_unmapped_mate function checks if the current read should be added to the unmapped_mate flag or start
+    creating a unmapped_mate flag.
+
+    :param read: pysam object containing data of a read.
+    :param flags: a 2d list containing all the flag information.
+    :param isbuildingflags: a list indicating which flags are currently being built.
+    :param all_flags: A 2d list containing the coordinates of the flags and additional information.
+    :param chromosome: The chromosome where the read is mapped.
+    :param start: Integer indicating the starting position of the read.
+
+    :return flags: a 2d list containing all the flag information.
+    :return isbuildingflags: a list indicating which flags are currently being built.
+    :return all_flags: A 2d list containing the coordinates of the flags and additional information.
+    """
+    if read.mate_is_unmapped:
+        flags, isbuildingflags = generate_flag(read, flags, isbuildingflags, 2)
+
+    elif not read.mate_is_unmapped and isbuildingflags[2] and start > flags[2][2]:
+        percentage = round(flags[2][3]['count'] / flags[2][3]['total'], 2)
+        if flags[2][3]['count'] > args.threshold and percentage > args.minpercentage:
+            all_flags.append(flags[2])
+        flags[2] = [chromosome, None, None, {'type': 'unmapped_mate', 'count': 0, 'total': 0}]
+        isbuildingflags[2] = False
+
+    return flags, isbuildingflags, all_flags
+
+
+def update_total(flags, isbuildingflags):
+    """ The update_total function iterates over all the flag types and increments the total number of reads it has
+    encountered by 1.
+
+    :param flags: a 2d list containing all the flag information.
+    return flags: a 2d list containing all the flag information.
+    """
+    for index in range(0, len(flags)):
+        if isbuildingflags[index]:
+            flags[index][3]['total'] += 1
```

Comment:
`enumerate` gebruiken?

Reply:
Ik itereer hier over 2 lists tegelijk en als de index in de ene list true is dan moet die de integer met 1 verhogen in de andere list dus enumerate is hier niet praktisch.

Reply:
Met enumerate kan je dit doen:

```python
for index, value in enumerate(flags):
    if isbuildingflags[index]:
        flags[index][3]['total'] += 1
```